contracts/service-registry/src/state.rs

```diff
@@ -134,6 +134,25 @@ pub fn service(
     }
 }
 
+pub fn services(
+    storage: &dyn Storage,
+    service_name: Option<&ServiceName>,
+    limit: nonempty::Usize,
+) -> error_stack::Result<Vec<Service>, ContractError> {
+    let start = service_name.map(Bound::exclusive);
+
+    let services: Vec<_> = SERVICES
+        .range(storage, start, None, Order::Ascending)
+        .filter_map(|res| match res {
+            Ok((_, res)) => Some(res),
+            _ => None,
+        })
+        .take(limit.into())
+        .collect();
+
+    Ok(services)
```

Comment:
either you return the error from the iteration, or you remove the result, but right now this function cannot fail